sgdp4/tle.c

```diff
@@ -153,7 +153,8 @@ su_orbit_parse_tle_line(orbit_t *self, unsigned int num, const char *linebuf)
        */
       n = sscanf(
         linebuf,
-        "%1u %05u %8lf %8lf %07u %8lf %8lf %11lf%5[0-9 ]%1u",
+        linebuf[52] != ' ' ? "%1u %05u %8lf %8lf %07u %8lf %8lf %11lf%5[0-9 ]%1u" : 
```

Comment:
This looks a bit too rigged up to my taste. We should focus on providing a sscanf format that does the work right and, if it was not possible, unpack the values in a different way.

Reply:
Also this may crash if `strlen(linebuf) <= 52`